pkg/fanal/artifact/image/image.go

```diff
@@ -359,25 +358,25 @@ func (a Artifact) isCompressed(l v1.Layer) bool {
 }
 
 func (a Artifact) inspectConfig(imageID string, osFound types.OS) error {
-	configBlob, err := a.image.RawConfigFile()
+	config, err := a.image.ConfigFile()
 	if err != nil {
 		return xerrors.Errorf("unable to get config blob: %w", err)
 	}
 
-	pkgs := a.analyzer.AnalyzeImageConfig(osFound, configBlob)
+	result := lo.FromPtr(a.analyzer.AnalyzeImageConfig(osFound, config))
 
-	var s1 v1.ConfigFile
-	if err = json.Unmarshal(configBlob, &s1); err != nil {
-		return xerrors.Errorf("json marshal error: %w", err)
+	var historyPkgs types.Packages
+	if len(result.PackageInfos) == 1 {
```

Comment:
Wht do we need this check?
Is it because we only have 1 `configAnalyzer`?

If i understand correctly - in the future we can have more then one `PackageInfos`(we append them in [merge](https://github.com/knqyf263/trivy/blob/2d3aedeecf539f221e3d73048dedb1d9071bec9e/pkg/fanal/analyzer/analyzer.go#L259-L261) func)
If yes - maybe we can create func to merge these packages right away?

Reply:
The length should be 0 or 1 as we just need to take packages detected from the image history. But I refactored a bit.
https://github.com/aquasecurity/trivy/pull/3434/commits/9858ad2c9dfe164f91cad33e3dd518745bc50f87